src/ServiceControl/Infrastructure/RavenDB/Expiration/AuditMessageCleaner.cs

```diff
@@ -0,0 +1,125 @@
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using NServiceBus.Logging;
+    using Raven.Abstractions;
+    using Raven.Abstractions.Commands;
+    using Raven.Abstractions.Data;
+    using Raven.Database;
+    using Raven.Json.Linq;
+
+    public static class AuditMessageCleaner
+    {
+        static ILog logger = LogManager.GetLogger(typeof(AuditMessageCleaner));
+
+        public static void Clean(int deletionBatchSize, DocumentDatabase database, DateTime expiryThreshold)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            var items = new List<ICommandData>(deletionBatchSize);
+            var attachments = new List<string>(deletionBatchSize);
+            try
+            {
+                var query = new IndexQuery
+                {
+                    Start = 0,
+                    PageSize = deletionBatchSize,
+                    Cutoff = SystemTime.UtcNow,
+                    DisableCaching = true,
+                    Query = string.Format("Status:[3 TO 4] AND ProcessedAt:[* TO {0}]", expiryThreshold.Ticks),
+                    FieldsToFetch = new[]
+                    {
+                        "__document_id",
+                        "MessageMetadata"
+                    },
+                    SortedFields = new[]
+                    {
+                        new SortedField("ProcessedAt")
+                        {
+                            Field = "ProcessedAt",
+                            Descending = false
+                        }
+                    }
+                };
+                var indexName = new ExpiryProcessedMessageIndex().IndexName;
+                database.Query(indexName, query, database.WorkContext.CancellationToken,
+                    null,
+                    doc =>
+                    {
+                        var id = doc.Value<string>("__document_id");
+                        if (string.IsNullOrEmpty(id))
+                        {
+                            return;
+                        }
+
+                        items.Add(new DeleteCommandData
+                        {
+                            Key = id
+                        });
+
+                        string bodyId;
+                        if (TryGetBodyId(doc, out bodyId))
+                        {
+                            attachments.Add(bodyId);
+                        }
+                    });
+            }
+            catch (OperationCanceledException)
+            {
+                //Ignore
+            }
+
+            var deletionCount = 0;
+
+            Chunker.ExecuteInChunks(items.Count, (s, e) =>
+            {
+                logger.InfoFormat("Batching deletion of {0}-{1} audit documents.", s, e);
+                var results = database.Batch(items.GetRange(s, e - s + 1));
+                logger.InfoFormat("Batching deletion of {0}-{1} audit documents completed.", s, e);
```

Comment:
This will look weird:

> Batching deletion of 500-999 audit documents completed.

Sounds like we aren't sure how many documents we deleted.